where-filter.js

```diff
@@ -230,6 +230,20 @@ function whereFilter(where) {
 			if (Array.isArray(value)) {
 				const matcher = where[key];
 
+				if (matcher.some) {
+					return value.some(function(v) {
+						return whereFilter(matcher.some)(v);
+					});
+				}
+				if (matcher.all) {
+					return value.every(function(v) {
+						return whereFilter(matcher.all)(v);
+					});
+				}
```

Comment:
So these can nest, right?

I guess we can't ever have a property called _some_ or _all_ as part of job data now?

```
{
			lines: {
				some: {
					data: {
              all: {
                  value: 'good'
              },
          },
				},
			},
		};
```

Reply:
If they can nest, which I think they can, it would be good to have a test for that.

Reply:
This does not collide with properties of "some" or "all". Without these changes, querying on an array property (or database analog - related entities) wouldn't really make sense. These are only treated "special" when they are in the value of a key that is an array. Just like how "eq" and "inq" do not preclude you from from using those as property names

Reply:
You can still use "some" and "all" as object keys. They only function in a "special" way when they are applied to an array property. Much like how use of "inq" and "eq" don't stop you from using those as property names. They only kick in as "special" when they are used on certain contexts where they only makes sense as special

Reply:
Ahh.. that's good to know.